src/codec_08004544.c

```diff
@@ -53,3 +53,30 @@ u16 sub_08004614(u32 r0, u32 r1, u32 r2, u32 r3, u32 sp10, u32 sp14, u32 sp18)
     sp10 <<= 0x10;
     return  v1 | sp14 | sp18 | (r3 >> 8 | sp10 >> 14);
 }
+
+u32 sub_0800463C(u16 r0)
+{
+    register u32 r2 asm("r2") = gUnk_03002990[0] & (0x100 << r0); // TODO:fix fakematch
+    return (r2 | -r2) >> 0x1f;
```

Comment:
Makes zero sense. 
(var | -var) >> 0x1f is just an optimization for (var != 0).